src/sp/it/pl/util/system/Environment.kt

```diff
@@ -47,96 +47,105 @@ fun copyToSysClipboard(s: String?) = copyToSysClipboard(DataFormat.PLAIN_TEXT, s
+            Try.error<Process, Exception>(e)
         }
     }
 }
 
-/**
- * Browse the file in the systems file browser by opening the parent directory and selecting it.
- * If it is unsupported, it will try to [open] the parent directory instead.
- */
-fun File.browse() {
-    if (Os.WINDOWS.isCurrent) {
-        openWindowsExplorerAndSelect()
-    } else {
-        // Would be nice if this was widely supported, but it isn't
-        if (Desktop.Action.BROWSE_FILE_DIR.isSupportedOrWarn()) {
-            Desktop.getDesktop().browseFileDirectory(this)
-        } else {
-            parentDirOrRoot.open()
-        }
-    }
-}
+/** Equivalent to [URI.browse] for the URI denoting this file. */
+fun File.browse() = toURI().browse()
 
 /**
- * Browse uri:
- * - directory/file -> calls [File.browse]
- * - url -> opens it in system browser
+ * Browse this uri, in order:
+ * - if it is directory/file -> browse the file in the system file browser by opening the parent directory and selecting the file.
+ * If this is unsupported, [open] is called on the parent directory.
+ * - if it is url -> open the url in system browser
  *
  * On some platforms the operation may be unsupported. In that case this method is a no-op.
  */
 fun URI.browse() {
     logger.info { "Browsing uri=$this" }
     runNew {
-        val browse = toFileOrNull()?.browse()
-        if (browse==null) {
-            try {
-                if (Desktop.Action.BROWSE.isSupportedOrWarn())
-                    Desktop.getDesktop().browse(this)
-            } catch (e: IOException) {
-                logger.error(e) { "Browsing uri=$this failed" }
-            }
-        }
+        toFileOrNull()
+                .ifNotNull {
+                    if (Os.WINDOWS.isCurrent) {
+                        it.openWindowsExplorerAndSelect()
+                    } else {
+                        // Would be nice if this was widely supported, but it isn't
+                        if (Desktop.Action.BROWSE_FILE_DIR.isSupportedOrWarn()) {
+                            Desktop.getDesktop().browseFileDirectory(it)
+                        } else {
+                            it.parentDirOrRoot.open()
+                        }
+                    }
+                }
+                .ifNull {
+                    try {
+                        if (Desktop.Action.BROWSE.isSupportedOrWarn())
+                            Desktop.getDesktop().browse(this)
+                    } catch (e: IOException) {
+                        logger.error(e) { "Browsing uri=$this failed" }
+                    }
+                }
     }
 }
 
 /**
- * Edit the file, in order:
- * - directory -> [open] is called
- * - file -> open in system associated editor
- *           if there is no association or the edit action is not supported, then calls [open] instead
- * - does not exist -> no-op
+ * Edit this file, in order:
+ * - if it is directory -> [open] is called
```

Comment:
why did you add "if it is" back in everywhere? It is just distracting, I had a reason to make it so concise...

Reply:
Because it is was not proper grammar and the subject was not clear. It was frankly very confusing. ```file -> ...``` File what? This is documentation. I understand the 'lambda' notation borrowed from Kotlin's when expression, but hiding the subject is going too far. Also, URI is not file or directory, it represents it and the correct word in the context is 'denote' not 'is'. This follows javadoc from JDK and is the most clear and concise way to explain. ```If this denotes X then``` should be used throughout the documentation in my opinion. Not ```file -> ...```

That's also why I added ```Open/edit/browse THIS file, in order:``` everywhere. We must properly disambiguate what this and it mean in the javadoc. 'It' must only be used after 'this' was used in previous sentence.

I would normally suggest to just write
```
If this denotes:
file -> ...
directory -> ...
```
but there are cases based on non denotative tests, like whether action is supported or whether file exists so I would rather avoid this version as well.

Reply:
```
If this denotes:
file -> ...
directory -> ...
if this does not exist -> ...
```
Why not like this? Clear and concise

Reply:
Yes, I actually copied your example and added that clause precisely because I read your reply. I however think that it looks fine this way. I liked the concise appearance of it, but it is not super important tbh

Reply:
It is fine, lets roll with it and see after some time.